tomoxlending/order_processor.go

```diff
@@ -861,62 +861,58 @@ func (l *Lending) GetCollateralPrices(header *types.Header, chain consensus.Chai
+		collateralPrice = new(big.Int).Div(collateralPrice, lendingTokenDecimal)
+		return lendTokenTOMOPrice, collateralPrice, nil
+	}
 	// if contract doesn't provide any price information
-	// getting price from direct pair in tomox
+	// getting price from pair in tomox
 	lastAveragePrice, err := l.GetMediumTradePriceBeforeEpoch(chain, statedb, tradingStateDb, collateralToken, lendingToken)
 	if err != nil {
-		return lendTokenTOMOPrice, collateralPriceFromContract, err
+		return nil, nil, err
 	}
 	if lastAveragePrice != nil && lastAveragePrice.Sign() > 0 {
-		log.Debug("Getting collateralPrice from direct pair in tomox", "lendToken", lendingToken.Hex(), "collateralToken", collateralToken.Hex(), "price", lastAveragePrice)
+		log.Debug("Getting collateral/lending from direct pair in tomox", "lendToken", lendingToken.Hex(), "collateralToken", collateralToken.Hex(), "price", lastAveragePrice)
 		return lendTokenTOMOPrice, lastAveragePrice, nil
 	}
 
-	var collateralPrice, collateralTOMOPrice *big.Int
-	collateralTOMOBasePriceFromContract, collateralTOMOUpdatedBlock := lendingstate.GetCollateralPrice(statedb, collateralToken, common.HexToAddress(common.TomoNativeAddress))
-	collateralTOMOBasePriceUpdatedFromContract := collateralTOMOUpdatedBlock.Uint64()/chain.Config().Posv.Epoch == header.Number.Uint64()/chain.Config().Posv.Epoch
-	if collateralToken == common.HexToAddress(common.TomoNativeAddress) {
-		collateralTOMOPrice = common.BasePrice
-	} else if collateralTOMOBasePriceUpdatedFromContract {
-		// getting collateralToken price from contract first
-		// otherwise, getting from tomox collateralToken/TOMO
-		log.Debug("Getting collateralTOMOPrice from contract", "collateralPrice", collateralPrice)
-		collateralTOMOPrice = collateralTOMOBasePriceFromContract
-	} else {
-		collateralTOMOPrice, err = l.GetMediumTradePriceBeforeEpoch(chain, statedb, tradingStateDb, collateralToken, common.HexToAddress(common.TomoNativeAddress))
-		if err != nil {
-			return collateralPrice, lendTokenTOMOPrice, err
-		}
-		log.Debug("Getting collateralTOMOPrice from tomox", "collateralTOMOPrice", collateralTOMOPrice, "err", err)
+	lastInverseAveragePrice, err := l.GetMediumTradePriceBeforeEpoch(chain, statedb, tradingStateDb, lendingToken, collateralToken)
+	if err != nil {
+		return nil, nil, err
 	}
-	lendingTokenDecimal, err := l.tomox.GetTokenDecimal(chain, statedb, common.Address{}, lendingToken)
-	log.Debug("GetTokenDecimal", "lendingToken", lendingToken, "err", err)
+	if lastInverseAveragePrice != nil && lastInverseAveragePrice.Sign() > 0 {
+		log.Debug("Getting collateral/lending from inverse pair in tomox", "lendToken", lendingToken.Hex(), "collateralToken", collateralToken.Hex(), "price", lastAveragePrice)
+		collateralPrice = new(big.Int).Mul(lastInverseAveragePrice, collateralTokenDecimal)
+		collateralPrice = new(big.Int).Div(collateralPrice, lendingTokenDecimal)
+		return lendTokenTOMOPrice, collateralPrice, nil
+	}
```

Comment:
not necessary 
inversePrice is calculated in  https://github.com/tomochain/tomochain/pull/255/files#diff-bb070402da3cf4811b1d48bdf02391e6R896

Reply:
if not found inverse price from smart contract try get it from tomox

Reply:
removed